### ISSUE TITLE
Regression: auth.proxy no longer allows org and role to be changed after initial user creation

### ISSUE THREAD

**Issue by candlerb** [grafana/grafana#59282]
**What happened**:

I upgraded from grafana 8.4.3 to 9.1.6 (and then subsequently to 9.2.6, to check that this later version doesn't fix the problem)

I am using auth.proxy with the configuration below.  How this used to work was: users who connect for the first time would be automatically created in org 11 ("Holding Pen"), which has no dashboards. I could then manually assign them to an appropriate organization and role, via the admin interface in Grafana.

But now, Grafana does not let me change either the organization or the role for users once they have been created.

* Under `/admin/users/edit/45`:
    * Change Role is dimmed out. A question-mark help bubble says: _"This user's role is not editable because it is synchronized from your auth provider. Refer to the Grafana authentication docs for details"_
    * I can remove the user from this organization, but there's no way to add them to another organization.
* Under `/admin/orgs/edit/11`
    * If I hover over the Role it shows a "no entry" sign
    * No way to move them to another organization.

The only thing I can do is to make them a full Grafana Admin, which I don't want to do for obvious reasons.

**What you expected to happen**:

Functionality to work as before: once a user has been seen for the first time, and auto-created within a given org and role, then as an admin I should be able to edit them by adding them to one or more other organizations, and set their role in each organization.

I did notice the new setting `oauth_skip_org_role_update_sync = true`, and I have set it just in case that helps, but I'm using auth.proxy rather than any of the oauth types.

I suspect what's happened is that a new setting is required, similar to `oauth_skip_org_role_update_sync = true`, for auth.proxy logins.  Currently, it seems that all auth.proxy users are forced to remain on their default org and role forever.

**How to reproduce it (as minimally and precisely as possible)**:

```
[users]
allow_sign_up = false
allow_org_create = false
auto_assign_org = true
auto_assign_org_id = 11
auto_assign_org_role = Viewer

[auth]
disable_login_form = true
disable_signout_menu = true
oauth_skip_org_role_update_sync = true

[auth.basic]
enabled = false

[auth.proxy]
enabled = true
header_name = OIDC_CLAIM_email
header_property = email
auto_sign_up = true
sync_ttl = 60
whitelist = 10.85.179.1
headers = Name:OIDC_CLAIM_name
```

10.85.179.1 is the address of my Apache proxy, which is using mod_auth_openidc to authenticate against Google.

I could try switching to Grafana's auth.google, but since the same proxy is authenticating a number of services, it would involve significant changes.

**Anything else we need to know?**:

Possibly related:
* #8816
* #19943

However, I don't actually *want* to set the org from a proxy header.  I liked it how it worked before: I was able to manage users via the Grafana admin interface, and I was able to make a user a member of multiple organizations this way.

I could live without the multiple org memberships though.  What's critical is being able to assign a user to at least *one* organization of my choice, with the appropriate role.

-----

My apt history shows that this server has been through a lot of grafana versions: 5.1.3, 6.1.0, 6.1.6, 6.3.2, 6.3.3, 6.3.5, 6.5.0, 6.6.2, 7.1.5, 7.3.7, 8.0.4, 8.1.2, 8.1.8, 8.4.3, 9.1.6, 9.2.6.

8.4.3 was installed on 2022-03-10.  I can't be *100%* sure it was the 8.4.3->9.1.6 upgrade which broke auth.proxy; it is *possible* that it was one of the earlier upgrades and nobody noticed it until now (users don't change often here). But it was definitely broken in 9.1.6, and remains broken in 9.2.6.  I'd rather not rollback to an ancient version now, just to test out exactly where it broke.

**Environment**:
- Grafana version: 9.2.6
- Data source type & version: n/a
- OS Grafana is installed on: Debian 9.13 
- User OS & Browser: macOS, chrome
- Grafana plugins: None
- Others: n/a

**Comment by candlerb** [comment]
Changing from auth.proxy to auth.google has worked around the problem (unless it [changes in Grafana 10](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/#role-mapping)).

Any users who last logged in with "Auth Proxy" cannot have their organization membership or role changed.  But once they login using Google, their auth type flips to "Google":

![image](https://user-images.githubusercontent.com/44789/203776598-a18c4bad-076a-42d1-8b37-89977ff611ba.png)

...and then at `/admin/users/edit/NN` the button "Add user to organization" returns, and "Change role" next to existing orgs is no longer dimmed out.

**Comment by usmangt** [comment]
Hi @candlerb,

Thanks for opening this issue and also providing very good detailed information.

I will forward this issue to our Engineers as they might be able to tell in more detail.

**Comment by gamab** [comment]
Merged and backported. Fix will be available in `9.3.1` and `9.2.8`.
